### ISSUE TITLE
Tiles-like artifacts

### ISSUE THREAD

**Issue by nishida-naoki** [wanmeihuali/taichi_3d_gaussian_splatting#119]
@wanmeihuali 

I found tiles-like artifacts when visualizing some parquet files, as shown in the attached image. 

After my short inspection, I found that `gaussian_alpha` of the following lines takes value of inf in these pixels, which leads to the occupation of whole one tile by one color.
https://github.com/wanmeihuali/taichi_3d_gaussian_splatting/blob/main/taichi_3d_gaussian_splatting/GaussianPointCloudRasterisation.py#L403-L407

Do you have any idea on this issue or any suggestion for further inspection?

To reproduce the issue, please download the attached parquet file and run `python3 visualizer.py --parquet_path_list refined.parquet` .

By the way, thank you for sharing your great project! 😄 

[parquet.zip](https://github.com/wanmeihuali/taichi_3d_gaussian_splatting/files/12518168/parquet.zip)

![Screenshot from 2023-09-05 11-36-17](https://github.com/wanmeihuali/taichi_3d_gaussian_splatting/assets/75428378/acb0d770-dfcd-46f1-899f-e85fdcc0a199)

**Comment by wanmeihuali** [comment]
This seems to be a numerical stability issue. Perhaps small values have occurred in some places, leading to problems like division by zero.

Strategy One: One approach is to directly remove these points from the parquet file. Using pandas should make this process straightforward. A more elegant method would be to eliminate these points in the parquet-saving logic. After all, these points, whenever they appear, will affect the inference results, so we shall always remove them if they appear.

Strategy Two: Directly remove points with nan/inf values during training. The advantage of this approach is that it can mitigate the impact of these inf points by training. The current program handles the nan situation here: https://github.com/wanmeihuali/taichi_3d_gaussian_splatting/blob/f7631e327d3e6e995324f1755bc3da25d603a584/taichi_3d_gaussian_splatting/GaussianPointAdaptiveController.py#L205, where it periodically removes points with a nan alpha value. However, it seems I may have forgotten to consider the inf scenario. Could you try using torch.isinf and employ a method similar to the code referenced above to remove points with inf values?

Strategy Three: Locate areas where division by zero might be occurring. The calculation for alpha grad is here: https://github.com/wanmeihuali/taichi_3d_gaussian_splatting/blob/f7631e327d3e6e995324f1755bc3da25d603a584/taichi_3d_gaussian_splatting/GaussianPointCloudRasterisation.py#L605. I just revisited it and couldn't immediately identify places where division by zero might happen. The adapter controller doesn't modify alpha either. we may need to do breakpoint debugging to locate this bug... I'll try to look into it tomorrow or when I have time.

**Comment by wanmeihuali** [comment]
Hi @nishida-naoki, how do you set the threshold to conic? I just noticed that the div by zero can happen when computing the conic: https://github.com/wanmeihuali/taichi_3d_gaussian_splatting/blob/f7631e327d3e6e995324f1755bc3da25d603a584/taichi_3d_gaussian_splatting/utils.py#L255
Have you tried to prevent det_cov from being zero? e.g. if (ti.math.abs(det_cov) < 1e-5) det_cov = ti.math.sign(x) * 1e-5.

**Comment by jb-ye** [comment]
@wanmeihuali I encounter the same issue. It actually happens quite frequently based on a few examples I tried. Do you have a pointcloud file that re-produces the bug?

@nishida-naoki Do you have a fix for your example?

**Comment by nishida-naoki** [comment]
@jb-ye Hi,
For better visualization, insert two lines to [here](https://github.com/wanmeihuali/taichi_3d_gaussian_splatting/blob/030e5abf2667432decd6d707704162fb1cd4544c/taichi_3d_gaussian_splatting/GaussianPointCloudRasterisation.py#L447) was enough for me:
```python
+                if abs(gaussian_alpha) >= np.inf:
+                    continue
```

But filtering invalid gaussians out while training seems not to be so simple, because `abs(gaussian_alpha) >= np.inf` does not always mean invalid `inv_cov` or other intermediate variables.

**Comment by jb-ye** [comment]
We can close this issue as the root cause is fixed in the latest PR #153

**Comment by nishida-naoki** [comment]
Thanks a lot for your help! @jb-ye  @wanmeihuali
